ejb3/src/main/java/org/jboss/as/ejb3/security/RunAsPrincipalInterceptor.java

```diff
@@ -35,16 +39,30 @@
 import org.wildfly.security.auth.server.SecurityDomain;
 import org.wildfly.security.auth.server.SecurityIdentity;
 import org.wildfly.security.authz.AuthorizationFailureException;
+import org.wildfly.security.manager.WildFlySecurityManager;
 
 /**
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public class RunAsPrincipalInterceptor implements Interceptor {
     public static final String ANONYMOUS_PRINCIPAL = "anonymous";
+
     private final String runAsPrincipal;
+    private final boolean securityRequired;
 
-    public RunAsPrincipalInterceptor(final String runAsPrincipal) {
+    public RunAsPrincipalInterceptor(final String runAsPrincipal, boolean securityRequired) {
         this.runAsPrincipal = runAsPrincipal;
+        this.securityRequired = securityRequired;
+    }
+
+    private boolean remotingContextIsSet (){
```

Comment:
@darranl  Not sure if I should leave this method here or I should create for example SecurityActions class helper for this package or reuse some existing ones (i.e. make them public)?